brion/blueConfig.cpp

```diff
@@ -87,12 +87,17 @@ class BlueConfig
 public:
     explicit BlueConfig(const std::string& source)
     {
-        std::ifstream file(source.c_str());
-        if (!file.is_open())
-            LBTHROW(
-                std::runtime_error("Cannot open BlueConfig file " + source));
         std::stringstream buffer;
-        buffer << file.rdbuf();
+        if (boost::filesystem::is_regular_file(source))
+        {
+            std::ifstream file(source.c_str());
+            if (!file.is_open())
+                LBTHROW(std::runtime_error("Cannot open BlueConfig file " +
+                                           source));
+            buffer << file.rdbuf();
+        }
+        else
+            buffer << source;
```

Comment:
I don't like this implementation very much because it will make errors about inexistent files misleading. What about another constructor that takes a char*?

Reply:
I didn't want to advertise this as a real feature, hence hiding it behind the existing API was fine enough for me. It's purpose is not very useful anyways, as the BlueConfig just redirects to other paths. It's nice for a show-case, but in reality you have to open the file close to where all the other referenced entities reside.
So I'm gonna close this then.

Reply:
Much more useful would be a remote file browser in Sauron ;)